src/app/home-page/toolbox/search/search.component.ts

```diff
@@ -1,25 +1,48 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { IconDefinition, faSearch } from '@fortawesome/free-solid-svg-icons';
 import { HomePageService } from '../../../core/services/home-page.service';
 import { UpdateCoursesMessageService } from 'src/app/core/services/update-courses.service';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/internal/operators';
 
 @Component({
   selector: 'app-search',
   templateUrl: './search.component.html',
   styleUrls: ['./search.component.less']
 })
-export class SearchComponent {
+export class SearchComponent implements OnInit {
+
   public faSearchIcon: IconDefinition = faSearch;
-  public searchResult: string = '';
+  public searchInput: string = '';
+  public searchInputChange: Subject<string> = new Subject();
 
   constructor (
     private homePageService: HomePageService,
     private updateCoursesMessage: UpdateCoursesMessageService
   ) { }
 
-  public searchCourses(): void {
-    this.homePageService.searchCourse(this.searchResult).subscribe(
-      data => this.updateCoursesMessage.updateCourse(data)
-    );
+  public ngOnInit(): void {
+    this.searchInputChange
+      .pipe(debounceTime(1000), distinctUntilChanged())
+      .subscribe(result => {
+        this.searchInput = result;
+        this.getDataFromApi();
```

Comment:
implement this functionality using mergeMap

Reply:
completed